### DISCUSSION TITLE
Can I use Sunshine on a VM without GPU?

### DISCUSSION THREAD

**Discussion by finnim** [LizardByte/community#265] [Sunshine] [answered]
Hello, I was wondering if Sunshine is suitable to be used on linux virtual machines such as the ones created on proxmox. 
The server does not have a dedicated GPU and the CPU is a "server CPU" which does not have an integrated gpu. 

Would it be possible to run sunshine on a VM running on top of proxmox?

**Comment by ReenigneArcher** [comment, marked as answer]
Yes, it works if you use X11. Let me know if you have issues and I'll share my vm configs.

**Reply by gado4525** [reply]
Can You elaborate on how to set that ? I'm trying to set a win 11 vm with no gpu

**Reply by ReenigneArcher** [reply]
On Windows, you cannot.

**Reply by benhcai** [reply]
Would you be able to share how you did it? Trying to figure out the same thing rn!

**Reply by mousethatischarles** [reply]
Sunshine + Windows, no a VM..? ...yes, yes you can. (With limitations)
While looking for my own answers I came across this question that I do have an answer for, sort-of.
If your VM is running on an ESXi Host, or maybe any VMWare platform, do this:
1) Be sure you have VMWare Tools installed on your VM, then open your VM's settings:
2) Go to the 'Memory' tab and select 'Reserve all guest memory (All locked)
This is required to stop the next step throwing up errors and the VM failing to boot.
3) Go to the 'Video Card' tab, expand it, change 'Specify default settings' to 'Custom settings'. Then select 'Enable 3D Support' and leave 'Renderer' on 'Automatic'. There should be no need to fiddle with memory allocation at this point.
4) Save and try out your VM.
5) All being well the VM will start without errors and Sunshine should work normally without any changes to the baseline settings - hurrah!
6) After having shut down your VM go back and look at it's 'Video Card' tab - you will likely find it now says 16mb rather than 4mb. Don't change it unless it says 4mb, then change it to 16mb. The 256mb allocation as part of the 3D option can be increased, but I wouldn't bother unless you run in to problems.
7) The 3D renderer 'Auto-select' option will have likely changed to either 'Software' or 'Hardware' - expect the former but if the latter that's another hurrah! Regardless I would advise NOT fiddling, you will probably break it. Software rendering happens 'outside' your VM in the host system's time so it shouldn't be much of a problem.

Gotchas:
1) If you have a 'real' 3D card passed through to the VM you WILL find nothing inside the VM will use it anymore even though Windows claims otherwise, sigh. You WILL be stuck with VMWare's very slow 3D renderer. I have yet to find a solution for this, use Parsec instead and put up with software encoding from within the VM.
2) If you use a GFX card for VM passthrough be advised NVIDIA have been busily locking-down their 'datacentre' Tesla drivers to stop people using Tesla cards in VMs for GFX use, bastards! So if like me you have a Tesla card in your ESXi machine you're going to have to use GRID drivers instead - they can be made to work with a bit of fiddling but there are downsides such as no hardware encoding for the likes of Sunshine or Parsec - grrr!

For anyone who catches this reply I hope it helps.

**Reply by Matthias199** [reply]
@ReenigneArcher Yes please I whant to setup an Ubuntu VM with kodi and stream it with sunshine and moonlight to my samssung tizen TV

**Reply by ReenigneArcher** [reply]
@Matthias199 why not use Plex or Jellyfin for that?

**Reply by Matthias199** [reply]
@ReenigneArcher Plex is in use but i whant also some plugins like xstream :)

**Reply by Matthias199** [reply]
@ReenigneArcher  would you share your X11 setup for sunshine without GPU in a VM?

**Reply by ReenigneArcher** [reply]
@Matthias199 This is proxmox.

![image](https://github.com/LizardByte/Sunshine/assets/42013603/49868e9d-2847-4d21-9d28-a96d6e3e8877)
![image](https://github.com/LizardByte/Sunshine/assets/42013603/4bd413ff-0efa-44cc-ad49-6ae569b95eae)

**Reply by Matthias199** [reply]
ah okey no nothing special seems to work right out of the box i was thinking you have to emulate an GPU

**Reply by ShaoQ1ang** [reply]
> Can You elaborate on how to set that ? I'm trying to set a win 11 vm with no gpu

@trablin45 Hi，Have you found a solution? I also want to streaming vm with no gpu, but have no idea.
